Source/WebCore/Modules/WebGPU/Implementation/WebGPUComputePassEncoderImpl.cpp

```diff
@@ -71,16 +71,12 @@ void ComputePassEncoderImpl::end()
 void ComputePassEncoderImpl::setBindGroup(Index32 index, const BindGroup& bindGroup,
     std::optional<Vector<BufferDynamicOffset>>&& offsets)
 {
-    auto backingOffsets = valueOrDefault(offsets);
-    wgpuComputePassEncoderSetBindGroup(m_backing.get(), index, protectedConvertToBackingContext()->convertToBacking(bindGroup), static_cast<uint32_t>(backingOffsets.size()), backingOffsets.data());
+    wgpuComputePassEncoderSetBindGroup(m_backing.get(), index, protectedConvertToBackingContext()->convertToBacking(bindGroup), WTFMove(offsets));
 }
 
-void ComputePassEncoderImpl::setBindGroup(Index32 index, const BindGroup& bindGroup,
-    std::span<const uint32_t> dynamicOffsetsArrayBuffer,
-    Size64 dynamicOffsetsDataStart,
-    Size32 dynamicOffsetsDataLength)
+void ComputePassEncoderImpl::setBindGroup(Index32, const BindGroup&, std::span<const uint32_t>, Size64, Size32)
 {
-    wgpuComputePassEncoderSetBindGroup(m_backing.get(), index, protectedConvertToBackingContext()->convertToBacking(bindGroup), dynamicOffsetsDataLength, dynamicOffsetsArrayBuffer.subspan(dynamicOffsetsDataStart).data());
+    RELEASE_ASSERT_NOT_REACHED();
```

Comment:
Why is this needed but never called? Maybe worth a comment?

Reply:
The reasoning is `ComputePassEncoderImpl` (in GPUP) and `RemoteComputePassEncoderProxy` (in Web content process) both inherit from `ComputePassEncoder` (process agnostic) and only `RemoteComputePassEncoderProxy` supports both member functions as the specification does. But `RemoteComputePassEncoderProxy` translates both member functions into a single member function so the other one will never get called in the GPUP.

It evolved from when WebGPU was designed for WK1 and such a factoring was necessary, but since WebGPU is WK2 only now some of this outdated architecture remains